apps/meteor/tests/end-to-end/api/livechat/18-rooms-ee.ts

```diff
@@ -254,4 +254,76 @@ import { IS_EE } from '../../../e2e/config/constants';
 			expect(updatedRoom).to.not.have.property('onHold');
 		});
 	});
+
+	describe('visitor abandonment feature', () => {
+		let room: IOmnichannelRoom;
+
+		before(async () => {
+			await updateSetting('Livechat_abandoned_rooms_action', 'Livechat_close_chat');
+			await updateSetting('Livechat_visitor_inactivity_timeout', 60);
+		});
+
+		it('should set predictedVisitorAbandonmentAt when agent sends a message', async () => {
+			const { room: newRoom } = await startANewLivechatRoomAndTakeIt();
+
+			room = newRoom;
+
+			await sendAgentMessage(room._id);
+
+			const updatedRoom = await getLivechatRoomInfo(room._id);
+
+			const lastMessageTs = updatedRoom.responseBy?.lastMessageTs;
+			const firstResponseTs = updatedRoom.responseBy?.firstResponseTs;
+			const predictedVisitorAbandonmentAt = updatedRoom.omnichannel?.predictedVisitorAbandonmentAt;
+
+			expect(predictedVisitorAbandonmentAt).to.not.be.undefined;
+			expect(lastMessageTs).to.not.be.undefined;
+			expect(firstResponseTs).to.not.be.undefined;
+
+			if (!lastMessageTs || !predictedVisitorAbandonmentAt || !firstResponseTs) {
+				throw new Error('lastMessageTs, predictedVisitorAbandonmentAt and firstResponseTs should not be undefined');
+			}
+
+			// expect predictedVisitorAbandonmentAt to be 60 seconds after lastMessageTs
+			const lastMessageTsDate = new Date(lastMessageTs);
+			const predictedVisitorAbandonmentAtDate = new Date(predictedVisitorAbandonmentAt);
+			const firstResponseTsDate = new Date(firstResponseTs);
+
+			expect(predictedVisitorAbandonmentAtDate.getTime()).to.be.equal(lastMessageTsDate.getTime() + 60000);
+			expect(firstResponseTsDate.getTime()).to.be.equal(lastMessageTsDate.getTime());
+		});
+
+		it('should not update predictedVisitorAbandonmentAt when agent sends yet another message', async () => {
+			await sendAgentMessage(room._id);
+
+			const updatedRoom = await getLivechatRoomInfo(room._id);
+
+			const lastMessageTs = updatedRoom.responseBy?.lastMessageTs;
+			const firstResponseTs = updatedRoom.responseBy?.firstResponseTs;
+			const predictedVisitorAbandonmentAt = updatedRoom.omnichannel?.predictedVisitorAbandonmentAt;
+
+			expect(predictedVisitorAbandonmentAt).to.not.be.undefined;
+			expect(lastMessageTs).to.not.be.undefined;
+
+			if (!lastMessageTs || !predictedVisitorAbandonmentAt || !firstResponseTs) {
```

Comment:
Same as above, we can also add the `firstResponseTs` there as it's missing in the `expect`ations.

Reply:
https://github.com/RocketChat/Rocket.Chat/pull/30072#discussion_r1300925559